server/src/main/java/org/elasticsearch/discovery/PeerFinder.java

```diff
@@ -256,15 +258,20 @@ private Peer createConnectingPeer(TransportAddress transportAddress) {
     private boolean handleWakeUp() {
         assert holdsLock() : "PeerFinder mutex not held";
 
-        boolean peersRemoved = false;
-
-        for (final Peer peer : peersByAddress.values()) {
-            peersRemoved = peer.handleWakeUp() || peersRemoved; // care: avoid short-circuiting, each peer needs waking up
+        final List<TransportAddress> peersAddressesToRemove = new ArrayList<>();
+        for (final Entry<TransportAddress, Peer> addressAndPeer : peersByAddress.entrySet()) {
```

Comment:
maybe simpler to just write
```
final boolean peersRemoved = peersByAddress.values().removeIf(Peer::handleWakeUp);
```
and then further below just
```
return peersRemoved;
```

Reply:
Very slick. I pushed 96b02b0.